crates/kas-core/src/event/event.rs

```diff
@@ -61,22 +118,18 @@ pub enum Event<'a> {
     /// <kbd>Ctrl</kbd>, <kbd>Alt</kbd> or <kbd>Super</kbd> modifier keys are
     /// pressed. This is subject to change.
     Key(&'a KeyEvent, bool),
-    /// Input Method Editor: composed text changed
-    ///
-    /// Parameters are `text, cursor`.
+    /// An Input Method Editor event
     ///
-    /// This is only received after
+    /// IME events are only received after
     /// [requesting key focus](EventState::request_key_focus) with some `ime`
     /// purpose.
-    ImePreedit(&'a str, Option<(usize, usize)>),
-    /// Input Method Editor: composed text committed
     ///
-    /// Parameters are `text`.
-    ///
-    /// This is only received after
-    /// [requesting key focus](EventState::request_key_focus) with some `ime`
-    /// purpose.
-    ImeCommit(&'a str),
+    /// On [`Ime::Enabled`],
+    /// the widget should call [`EventState::set_ime_cursor_area`] immediately
+    /// and each time the area changes (relative to the widget's coordinate
+    /// space), until [`Ime::Disabled`] is received. Failure to do so will
+    /// result in the widget's entire `rect` being used as the IME cursor area.
+    Ime(Ime<'a>),
```

Comment:
All IME events are aggregated under `Event::Ime`; previously they weren't.

Motivation: a widget using IME should handle all IME events; other widgets can ignore them all.